### PR TITLE
Add notify user agent

### PR DESCRIPTION
**Pull request by diedpigs**
Basically re-write from Mike's ruby code `rc-useradd.rb`

### CODE REVIEW

File: notify_user.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        if record:
+            # Update counter
+            count = record['count']
+            table.update({'username': username, 'count': count + 1}, ['username'])
+
+            logger.debug(f'User {username} counter updated to {count + 1}')
+
+        else:
+            # Send email to user
+            receiver = [user_mail, mail_cfg.My_mail]
```
Comment (eesaanatluri) [inline notify_user.py]: I think this line should be `receiver = [user_email, mail_cfg.My_email]`

File: notify_user.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            message = Template(mail_cfg.Whole_mail).render(username=username, to=user_email)
+
+            if args.dry_run:
+                logger.info(f'smtp = smtplib.SMTP({mail_cfg.Server})')
+                logger.info(f'smtp.sendmail({sender}, {receiver}, message)')
```
Comment (eesaanatluri) [inline notify_user.py]: This should be `logger.info(f'smtp.sendmail({mail_cfg.Sender}, {receiver}, message)')`

File: notify_user.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+                logger.info(f"table.insert({'username': {username}, 'count': 1, 'sent_at': {datetime.now()}})")
```
Comment (eesaanatluri) [inline notify_user.py]: I got this error for the above format string `valueerror invalid format specifier`. Fixed it by doing this. You might want to put the row dict a few lines above. 
```
row = {'username': {username}, 'count': 1, 'sent_at': {datetime.now()}}
logger.info(f"table.insert({row})")
```

File: notify_user.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+            else:
+                smtp = smtplib.SMTP(mail_cfg.Server)
+                smtp.sendmail(sender, receiver, message)
```
Comment (eesaanatluri) [inline notify_user.py]: Similar to the change on line 50, this should be - `smtp.sendmail(mail_cfg.Sender, receiver, message)`

**Review by eesaanatluri — CHANGES_REQUESTED** [review]

File: notify_user.py (hunk 3 above)
Comment (diedpigs) [reply notify_user.py]: should be fixed in 86d9569

File: notify_user.py (hunk 4 above)
Comment (diedpigs) [reply notify_user.py]: should be fixed in 5ecfce5

File: notify_user.py (hunk 2 above)
Comment (diedpigs) [reply notify_user.py]: should be fixed in 5ecfce5

File: notify_user.py (hunk 1 above)
Comment (diedpigs) [reply notify_user.py]: should be fixed in b4d9663

File: notify_user.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        if record:
+            # Update counter
+            count = record['count']
+            table.update({'username': username, 'count': count + 1}, ['username'])
+
+            logger.debug(f'User {username} counter updated to {count + 1}')
+
+        else:
+            # Send email to user
+            receivers = [user_mail, mail_cfg.My_email]
```
Comment (eesaanatluri) [inline notify_user.py]: The `user_mail` in this line should be `user_email` actually. Can you please change it.

Comment (diedpigs) [reply notify_user.py]: Should be fix in 4be0752

**Pull request merged**